### PR TITLE
guard call to jnr Platform.getCPU via Native wrapper

### PR DESCRIPTION
**Pull request by tomekl007**

### CODE REVIEW

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java [hunk 1]
```diff
@@ -75,6 +77,32 @@ public static int getProcessId() {
...
+  /**
+   * Returns the current processor architecture the JVM is running on, as reported by {@link
+   * Platform#getCPU()}.
+   *
+   * @return the current processor architecture.
+   * @throws UnsupportedOperationException if JNR Platform library is not loaded.
+   */
+  public static String getCPU() {
+    if (!isPlatformAvailable())
+      throw new UnsupportedOperationException("JNR Platform class not loaded");
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: I see that similar methods in this class use `IllegalStateException`, shouldn't we do the same here?

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java [hunk 2]
```diff
@@ -150,4 +178,27 @@ private Timeval(Runtime runtime) {
...
+
+    static {
+      Platform platform;
+      try {
+        Class<?> platformClass = Class.forName("jnr.ffi.Platform");
+        Method getNativePlatform = platformClass.getMethod("getNativePlatform");
+        platform = (Platform) getNativePlatform.invoke(null);
+      } catch (Throwable t) {
+        platform = null;
+        if (LOG.isDebugEnabled())
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: I see that contrary to 3.x, in 4.x we only log at DEBUG level, again I think we should harmonize the code.

**Review by adutra — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java (hunk 1 above)
Comment (tomekl007) [reply core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: done

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java (hunk 2 above)
Comment (tomekl007) [reply core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: done

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java [hunk 3]
```diff
@@ -75,6 +77,34 @@ public static int getProcessId() {
...
+   * Platform#getCPU()}.
+   *
+   * @return the current processor architecture.
+   * @throws UnsupportedOperationException if JNR Platform library is not loaded.
+   */
+  public static String getCPU() {
+    if (!isPlatformAvailable())
+      throw new IllegalStateException(
+          "JNR Platform class not loaded. "
+              + "Check isGetProcessIdAvailable() before calling this method.");
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: Should be: "Check isPlatformAvailable()".

**Review by adutra — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java (hunk 3 above)
Comment (tomekl007) [reply core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: done, sorry for that copy/pasting

File: core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java [hunk 4]
```diff
@@ -75,6 +77,34 @@ public static int getProcessId() {
...
+      return false;
+    }
+  }
+
+  /**
+   * Returns the current processor architecture the JVM is running on, as reported by {@link
+   * Platform#getCPU()}.
+   *
+   * @return the current processor architecture.
+   * @throws UnsupportedOperationException if JNR Platform library is not loaded.
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java]: Should be `IllegalStateException`

**Review by adutra — CHANGES_REQUESTED** [review]

**Review by adutra — APPROVED** [review]

**Pull request merged**